core/optaplanner-constraint-streams/src/main/java/org/optaplanner/constraint/streams/bavet/BavetConstraintSessionFactory.java

```diff
@@ -57,7 +57,7 @@ public BavetConstraintSessionFactory(SolutionDescriptor<Solution_> solutionDescr
                 constraintMatchEnabled);
 
         Score_ zeroScore = scoreDefinition.getZeroScore();
-        Set<BavetAbstractConstraintStream<Solution_>> constraintStreamSet = new LinkedHashSet<>(constraintList.size() * 10);
+        Set<BavetAbstractConstraintStream<Solution_>> constraintStreamSet = new LinkedHashSet<>();
```

Comment:
Why take out the capacity hint? The size doesn't need to be accurate, just a good estimation is fine

Reply:
That was not intentional. Feel free to re-introduce it in your follow-up PR.